gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityServiceEntityFetcher.java

```diff
@@ -60,16 +63,19 @@ private EntityQueryRequest buildRequest(
 
     addGroupBys(exploreRequest, builder);
     addSelections(requestContext, exploreRequest, builder);
-
-    // Ideally, needs the limit and offset for group by, since the fetcher is only triggered when
-    // there is a group by, or a single aggregation selection. A single aggregated selection would
-    // always return a single result (i.e. limit 1)
-    builder.setLimit(DEFAULT_ENTITY_REQUEST_LIMIT);
-
-    // TODO: Push group by down to EQS
-    // If there is a group by, specify a large limit and track actual limit, offset and order by
-    // expression list, so we can compute these once the we get the results.
-    if (requestContext.hasGroupBy()) {
+    addLimitAndOffset(requestContext, exploreRequest, builder);
+
+    // TODO: Push order by down to EQS
+    // EQS (and document-store) currently doesn't support order by on functional expressions
+    // If there are order by expressions, specify a large limit and track actual limit, offset and
+    // order by
+    // expression list, so we can compute these once we get the results.
+    if (!requestContext.getOrderByExpressions().isEmpty()) {
```

Comment:
Can we move this code into addLimitAndOffset? It's confusing that we set the limit and offset unconditionally (in addLimitAndOffset), then conditionally overwrite them in the top level buildRequest call.

Also whether we push down or not will influence response processing. We should make sure we're in sync (i.e. have common code or state so we only evaluate the condition in one place in code).